ironfish/src/wallet/walletdb/walletdb.ts

```diff
@@ -740,12 +743,40 @@ export class WalletDB {
     }
   }
 
+  private loadNullifierBloomFilter(tx?: IDatabaseTransaction): Promise<BloomFilter> {
+    if (this.nullifierBloomFilter) {
+      return Promise.resolve(this.nullifierBloomFilter)
+    } else {
+      return this.db.withTransaction(tx, async (tx) => {
+        if (this.nullifierBloomFilter) {
+          // A concurrent call to this function already created the filter
+          return this.nullifierBloomFilter
+        }
+
+        const nullifierBloomFilter = new BloomFilter(0x800000) // 1 MiB bloom filter
+        for await (const [_accountPrefix, nullifier] of this.nullifierToNoteHash.getAllKeysIter(
+          tx,
+        )) {
+          nullifierBloomFilter.put(nullifier)
+        }
+
+        this.nullifierBloomFilter = nullifierBloomFilter
+        return nullifierBloomFilter
+      })
+    }
+  }
+
   async loadNoteHash(
     account: Account,
     nullifier: Buffer,
     tx?: IDatabaseTransaction,
   ): Promise<Buffer | undefined> {
-    return this.nullifierToNoteHash.get([account.prefix, nullifier], tx)
+    const nullifierFilter = await this.loadNullifierBloomFilter(tx)
```

Comment:
should we call `loadNullifierBloomFilter` when the wallet loads?

Reply:
Not all usages of the wallet/walletdb may require access to the notes/nullifiers, so I preferred to go with this lazy loading route. If someone has a lot of notes, this can *potentially* add a noticeable performance latency to every command/rpc call.

Reply:
That makes sense. Now that I think about it, calling this during `load` would be a major performance drag on every wallet command that a user runs without a running node.